src/core/lib/transport/metadata_batch.h

```diff
@@ -1314,6 +1358,20 @@ class MetadataMap {
     }
   }
 
+  // Sanitize the metadata map.
+  // Iterates over all encodable and unknown headers and calls the sanitize_fn
+  // for each of them. If the function returns true, the header is removed.
+  // If include_known_encodable_headers is false, then only the unknown headers
+  // are sanitized.
+  void Sanitize(metadata_detail::SanitizeFn sanitize_fn,
```

Comment:
I think we should also pass in an object here instead of `sanitize_fn` -- for built in types we can likely generate much better code here

Something like:

```
template <typename Filterer>
void Filter(Filterer* filterer) {
  table_.template FilterIn(...);
  unk_.erase(std::remove_if(...));
}
```

Reply:
Fixed now per your suggestions.